olp-cpp-sdk-dataservice-read/src/repositories/CatalogCacheRepository.cpp

```diff
@@ -111,6 +120,34 @@ boost::optional<model::VersionResponse> CatalogCacheRepository::GetVersion() {
   return boost::any_cast<model::VersionResponse>(cached_version);
 }
 
+void CatalogCacheRepository::PutVersionInfos(
+    std::int64_t start, std::int64_t end, const model::VersionInfos& versions) {
+  std::string hrn(hrn_.ToCatalogHRNString());
+  auto key = VersionInfosKey(hrn, start, end);
+  OLP_SDK_LOG_DEBUG_F(kLogTag, "PutVersionInfos -> '%s'", key.c_str());
+  auto list_versions = olp::serializer::serialize(versions);
+
+  auto versions_data = std::make_shared<std::vector<unsigned char>>(
+      std::begin(list_versions), std::end(list_versions));
+
+  cache_->Put(VersionInfosKey(hrn, start, end), versions_data, default_expiry_);
+}
+
+boost::optional<model::VersionInfos> CatalogCacheRepository::GetVersionInfos(
+    std::int64_t start, std::int64_t end) {
+  std::string hrn(hrn_.ToCatalogHRNString());
+  auto key = VersionInfosKey(hrn, start, end);
+  OLP_SDK_LOG_DEBUG_F(kLogTag, "GetVersionInfos -> '%s'", key.c_str());
+
+  auto value = cache_->Get(key);
+  if (!value) {
+    return boost::none;
+  }
+
+  auto list_versions = std::string(value->begin(), value->end());
+  return parser::parse<model::VersionInfos>(list_versions);
```

Comment:
Should be possible to add a new `parse` method, which takes in a `KeyValueCache::ValueTypePtr` or a `KeyValueCache::ValueType`.
Doesn't need to be in `core` for now it can just be in dataservice-read where the parsers are.